cmd/kubeadm/app/apis/kubeadm/v1beta2/types.go

```diff
@@ -23,119 +23,120 @@ import (
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// InitConfiguration contains a list of elements that is specific "kubeadm init"-only runtime
-// information.
+// InitConfiguration contains runtime information that are specific to "kubeadm init".
+// These information are only used the first time `kubeadm init` runs.
+// After that, the information in the fields IS NOT uploaded to the `kubeadm-config` ConfigMap
+// that is used by `kubeadm upgrade` for instance. These fields must be optional.
 type InitConfiguration struct {
 	metav1.TypeMeta `json:",inline"`
 
-	// `kubeadm init`-only information. These fields are solely used the first time `kubeadm init` runs.
-	// After that, the information in the fields IS NOT uploaded to the `kubeadm-config` ConfigMap
-	// that is used by `kubeadm upgrade` for instance. These fields must be omitempty.
-
-	// BootstrapTokens is respected at `kubeadm init` time and describes a set of Bootstrap Tokens to create.
-	// This information IS NOT uploaded to the kubeadm cluster configmap, partly because of its sensitive nature
+	// `bootstrapTokens` describes a set of Bootstrap Tokens to create during `kubeadm init`.
+	// This information is NOT uploaded to the `kubeadm-config` ConfigMap, partly because of its sensitive nature.
 	BootstrapTokens []BootstrapToken `json:"bootstrapTokens,omitempty"`
 
-	// NodeRegistration holds fields that relate to registering the new control-plane node to the cluster
+	// `nodeRegistration` holds fields related to registering the new control-plane node to the cluster.
 	NodeRegistration NodeRegistrationOptions `json:"nodeRegistration,omitempty"`
```

Comment:
as per goling, the comment above exported fields should start with the field name matched - i.e.

```
// NodeRegistration ...
NodeRegistration ...
```

currently golint is disabled on the kubeadm API packages due to other reasons, but we'd like to enable it at some point, and then it will fail if the golint spec is not followed.

Reply:
It depends.
If the comment is **only** for developers, putting `NodeRegistration` in the comment makes sense. Applying golint is also okay. If the comment is for generating docs for readers who don't care about the implementation, putting `NodeRegistration` in it will cause confusions. These readers only see the YAML files where fields are named as `caFile`, `apiServer`, `nodeRegistration`. They don't know why there a doc about `CAFile` while they need to put `caFile` in the YAML. They don't read Go source code and they probably never will.

Reply:
i'd like us to conform with golint on this aspect, which also opens the topic around consistency of user / developer docs.

```
// NodeRegistration ...
NodeRegistration ...
```

if we do that we can have all fields in the docs referenced with PascalCase (e.g. SomeField.SomeField) and on top of the document we can have a note that when the configuration is used from a YAML, JSON file all fields are represented in lowercase (camelCase). WDYT?

Reply:
No. I don't want to get myself into another debate ... really sick of these kind of discussions in the website. If the docs generated are incorrect, maybe we should fix `golint`. We have tools to serve people not people to make tools happy.

For a reference, kubelet config: https://github.com/kubernetes/kubelet/blob/master/config/v1beta1/types.go#L75